src/defck.rs

```diff
@@ -198,15 +191,7 @@ impl Database {
         self.parse_equality_commands(sset, definitions);
         let names = self.name_result();
 
-        // Fail the whole check if no equality has been defined
-        #[allow(clippy::manual_assert)]
-        if definitions.equalities.is_empty() {
-            definitions
-                .diagnostics
-                .insert(StatementAddress::default(), Diagnostic::DefCkNoEquality);
-        }
-
-        // TODO verify that the reflexivity, associativity, and transivity laws are well-formed
```

Comment:
Why did you remove this check?
I agree it's redundant with the `definitions.equalities.contains(&equality)` check further down, but
if no equality has been defined, all definitional checks will fail with the same error. 
In my opinion, in this case, it's better from the user point of view to get a single error message.

Reply:
It's not really the right check. You need one equality for each sort you are making definitions for (unless they have justification theorems), and so you want to report an error whenever a sort used for a definition doesn't have an equality for it. The natural way to encounter this issue is lazily on first occurrence of a definition which uses an unregistered sort, which basically amounts to deduplicating the `contains(&equality)` error you mention.

Reply:
Ok, we can also do it like that.